africastalking/Airtime.py

```diff
@@ -13,28 +12,41 @@ def _init_service(self):
 
     def send(self, phone_number=None, amount=None, currency_code=None, recipients=None, callback=None):
 
-        if recipients is not None:
-            def join_amount_and_currency(obj):
+        def join_amount_and_currency(obj):
                 obj['amount'] = " ".join([str(obj['currency_code']), str(obj['amount'])])
                 del obj['currency_code']
                 return obj
-            recipients = list(map(join_amount_and_currency, recipients))
-        if all(key is not None for key in [phone_number, amount, currency_code ]) and recipients is None:
-            amount = " ".join([str(currency_code), str(amount)])
-            recipients = [
-                {'phoneNumber': str(phone_number), 'amount': str(amount)},
-            ]
-
-        try:
-            schema = Schema([
-                {
-                    'phoneNumber': And(str, lambda s: validate_phone(s)),
-                    'amount': And(str, lambda s: validate_amount(s))
-                }
-            ])
-            recipients = schema.validate(recipients)
-        except SchemaError as err:
-            raise ValueError('Invalid recipients: ' + err.message)
+
+        def value_validator(phoneNumber, amount, currency_code):
+            if not validate_phone(phoneNumber):
+                return 'Invalid phone number'
+            elif not validate_amount(amount):
+                return 'Invalid amount' 
+            elif not validate_currency(currency_code):
+                return 'Invalid currency code'
+            else:
+                return False              
+
+        
+        if recipients is None:
+            recipients= [
+                {'phoneNumber': str(phone_number), 'amount': str(amount), 'currency_code': str(currency_code)}
+            ]            
+        
+        for recipient in recipients:
+            if not set(recipient.keys()) == {'phoneNumber', 'amount', 'currency_code'}:
```

Comment:
Declate the set as a gloabal util func that can be reused so that you just pass dict and keys and it returns a boolean and if possible the missing key. You can reuse this above.